netstat_poller.go

```diff
@@ -0,0 +1,76 @@
+package shh
+
+import (
+	"syscall"
+	"time"
+
+	"github.com/heroku/slog"
+	"github.com/shirou/gopsutil/net"
+)
+
+type Netstat struct {
+	measurements chan<- Measurement
+}
+
+func NewNetstatPoller(measurements chan<- Measurement) Netstat {
+	return Netstat{measurements: measurements}
+}
+
+func getStats() (map[string]int, error) {
+	conns, err := net.Connections("all")
+	if err != nil {
+		return nil, err
+	}
+
+	counts := make(map[string]int)
+	counts["UDP"] = 0
+	for _, conn := range conns {
+		if conn.Type == syscall.SOCK_DGRAM {
+			counts["UDP"]++
+			continue
+		}
+		c, ok := counts[conn.Status]
+		if !ok {
+			counts[conn.Status] = 0
+		}
+		counts[conn.Status] = c + 1
+	}
+
+	fields := map[string]int{
+		"tcp_established": counts["ESTABLISHED"],
+		"tcp_syn_sent":    counts["SYN_SENT"],
+		"tcp_syn_recv":    counts["SYN_RECV"],
+		"tcp_fin_wait1":   counts["FIN_WAIT1"],
+		"tcp_fin_wait2":   counts["FIN_WAIT2"],
+		"tcp_time_wait":   counts["TIME_WAIT"],
+		"tcp_close":       counts["CLOSE"],
+		"tcp_close_wait":  counts["CLOSE_WAIT"],
+		"tcp_last_ack":    counts["LAST_ACK"],
+		"tcp_listen":      counts["LISTEN"],
+		"tcp_closing":     counts["CLOSING"],
+		"tcp_none":        counts["NONE"],
+		"udp_socket":      counts["UDP"],
```

Comment:
What happens if one of these `counts[KEY_NAME]` is not present on the map? Are we confident these headings are the same across different distros and/or OSes?

Reply:
Good question, this library is providing this interface https://github.com/shirou/gopsutil - even if the value is 0 the count field and label is still there. These are all standard tcp connection state machine states so these should be consistent across all OSes and distros.

Reply:
There does appear to be subtle differences between the different statuses as you go between Oses.

ref: [windows](https://github.com/shirou/gopsutil/blob/e0ec1b9cda4470db704a862282a396986d7e930c/net/net_windows.go#L562-L576)
ref: [linux](https://github.com/shirou/gopsutil/blob/9a667f1188e49a92c6cabc345bd3c2b41da6513e/net/net_linux.go#L308-L321)

Reply:
In my opinion, we can focus on linux and leave windows aside. If someone really wants to run shh on windows, they are my guest to fix it if the netstat poller would become a problem.